### ISSUE TITLE
Eslint confused about indentation

### ISSUE THREAD

**Issue by krainboltgreene** [eslint/eslint#4360]
Despite being told I use 2 spaces for indentation it thinks I should be using 6 for a specific line?

``` json
{
  "rules": {
    "indent": [
      2,
      4
    ],
    "quotes": [
      2,
      "double"
    ],
    "linebreak-style": [
      2,
      "unix"
    ],
    "semi": [
      2,
      "never"
    ]
  },
  "env": {
    "es6": true,
    "browser": true
  },
  "ecmaFeatures": {
    "modules": true
  },
  "extends": "eslint:recommended"
}
```

``` javascript
import { createElement } from "react"
import { render } from "react-dom"
import { core } from "./admin/component"

export default class {
  constructor (state, target) {
    render(createElement(core, state), target)
  }
}
```

```
/lib/admin.js
  7:5  error  Expected indentation of 6 space characters but found 4  indent

✖ 1 problem (1 error, 0 warnings)
```

``` json
{
  // ...
  "eslint": "^1.9.0",
  // ...
}
```

**Comment by ilyavolodin** [comment]
That looks like a bug. CC @gyandeeps

**Comment by gyandeeps** [comment]
* @ilyavolodin We have not yet handled `classes` in indent rule. I am not sure if this should be flagged as a bug or enhancement.

* @krainboltgreene Based on your indent rule config
```js
"indent": [
      2,
      4
    ]
```

We are looking at 4 spaces and and not 2. To make it check for 2 spaces change the config to `[2, 2]`.

**Comment by IanVS** [comment]
@gyandeeps if the indent is set to 4 spaces, why would it be expecting 6?

**Comment by gyandeeps** [comment]
Remember, our indent rule measure indent relative to the parent, with that being said the constructor already has 2 indent so it expects the body of constructor to be indented with 4 spaces so 4 + 2 makes it 6.
Since our indent rule hasn't been updated to handle classes thats why its not checking class body indentation.

**Comment by ilyavolodin** [comment]
Closing as it sounds like the rule is doing the right thing (within it's limitations). @gyandeeps we should probably have an issue to track adding classes to indent.
